lib/spack/spack/cmd/clean.py

```diff
@@ -59,6 +60,9 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-p', '--python-cache', action='store_true',
         help="remove .pyc, .pyo files and __pycache__ folders")
+    subparser.add_argument(
+        '-l', '--locks', action='store_true',
+        help="remove lock files (typically unnecessary)")
     subparser.add_argument(
         '-a', '--all', action=AllClean, help="equivalent to -sdmp", nargs=0
```

Comment:
Can you add this to `--all`? You also need to add it to `any` below, otherwise it will also run `--stage`.